core/src/main/java/io/aiven/kafka/tieredstorage/RemoteStorageManager.java

```diff
@@ -146,13 +148,12 @@ public void configure(final Map<String, ?> configs) {
             );
             aesEncryptionProvider = new AesEncryptionProvider();
         }
-        chunkManager = new ChunkManager(
-            fetcher,
-            objectKey,
-            aesEncryptionProvider,
-            config.chunkCache()
-        );
-
+        final ChunkManagerFactory chunkManagerFactory = new ChunkManagerFactory();
+        chunkManagerFactory.configure(configs);
+        chunkManager = chunkManagerFactory.initChunkManager(fetcher,
+                                                            objectKey,
+                                                            aesEncryptionProvider,
+                                                            executor);
```

Comment:
Not sure if it's an issue, but shouldn't these executors be specific per component?

Reply:
So far the decision was made to use single executor. But we can reconsider.